packages/frontend-shared/src/utils/addExternalLinkClickListener.ts

```diff
@@ -0,0 +1,17 @@
+export const addExternalLinkClickListener = (mutation: { executeMutation: ({ url: string }) => void }) => {
+  document.addEventListener('click', (event: MouseEvent) => {
+    const target = event.target as HTMLElement
+    const anchor = target.closest('a[href]') as HTMLAnchorElement
+
+    if (anchor) {
+      const destination = anchor.href
+
+      if (destination.startsWith('/') || destination.startsWith('#') || destination.startsWith('http://localhost')) {
+        return
```

Comment:
While this function is pretty neat, I am not sure this is the best way to implement it. One notable problem is this can easily introduce a memory leak. 

This closure has a reference to a DOM node (target). Even after the DOM node is removed (eg, the component that created said DOM node is unmounted) the garbage collector will not be able to release the DOM node from memory, since it is still referenced in this event listener's closure. Although `App` probably won't be unmounted, it would only take a slight oversight for someone to use `addExternalLinkClickListener` in another component and we've have a very difficult time tracking down the memory leak.

One way to avoid this problem would be to have this composable clean up by using `onBeforeUnmount`:

```ts
const onClick = (event: MouseEvent) => {
  // ...
}

document.addEventListener('click', onClick)

onBeforeUnmount(() => {
  document.removeEventListener('click', onClick)
})
```

That said, rather than doing it this way, we might consider having a `<OpenExternalLink>` component? That would have all the required logic, so we wouldn't even need any global event listeners. That way, readers won't need to do the mental gymnastics to figure out how certain links open differently (by a global even listener in some distant component), the behavior would be clear and self contained in an `<OpenExternalLink>` component.

What do you think? I could be missing a compelling reason to do the `document.addEventListener` approach - let me know 👍

Reply:
One reason to get this into a PR quickly was to surface stuff like this as we had kind of informally discussed the approach a couple of times in slack threads, but hadn't really nailed it down.

The most compelling reason for me on the global listener is that when we're working, links will not need any special handling in order to do the thing that, 95% of the time, we want them to do. I did have some ideas for mitigating the risks you pointed out, and some others I can think of if you import this helper in the wrong place, but that's kind of moot as I agree that this makes sense to handle at the component level.

I think we now have enough reasons to create a `<BaseLink>` component. I'd been noticing there's a small number of classes that get added to every link. Especially if the `<BaseLink>` opens externally by default, since in reality most if not all internal links will be coming out of `<RouterLink>`. Then the ergonomics are about the same, if not better, since you get default styles and no event listener magic.